src/pages/CreateQuizPage.spec.tsx

```diff
@@ -0,0 +1,172 @@
+    vi.clearAllMocks();
+  });
+
+  it("should store quiz in local storage for guest users", async () => {
+    const { fillFields, submit } = setup(true);
+
+    await fillFields();
+    await submit();
+
+    const stored = JSON.parse(localStorage.getItem("guest_quizzes") || "[]");
+    expect(stored).toHaveLength(1);
+    expect(stored[0].title).toBe("test quiz");
+  });
+
+  it("should try to post quiz if user is authenticated", async () => {
+    const { fillFields, submit } = setup();
+
+    await fillFields();
+    await submit();
+
+    expect(screen.getByText(/test quiz/i)).toBeInTheDocument();
+  });
+
+  it("should show error if request fails", async () => {
+    const { fillFields, submit } = setup();
+    server.use(
+      http.post("/quizzes/", async () => {
+        return HttpResponse.error();
+      })
+    );
+
+    await fillFields();
+    await submit();
+
+    expect(toast.error).toHaveBeenCalled();
+    expect(screen.getByText(/wystąpił błąd/i)).toBeInTheDocument();
+  });
+
+  it("should show validation error if title is empty", async () => {
+    const { submit } = setup();
+
+    await submit();
+
+    expect(toast.error).toHaveBeenCalled();
+    expect(screen.getByText(/podaj tytuł/i)).toBeInTheDocument();
+  });
+
+  it("should show validation error if qustion field is empty", async () => {
+    const { user, submit } = setup();
+    await user.type(screen.getByPlaceholderText(/podaj tytuł/i), "test quiz");
+    await submit();
+
+    expect(toast.error).toBeCalled();
+    expect(screen.getByText(/pytanie.*treść/i));
+  });
+
+  it("should show validation error if one of the answer fields is empty", async () => {
+    const { user, submit } = setup();
+
+    await user.type(screen.getByPlaceholderText(/podaj tytuł/i), "test quiz");
+    await user.type(
+      screen.getByPlaceholderText(/treść pytania/i),
+      "test question"
+    );
+    await submit();
+
+    expect(toast.error).toBeCalled();
+    expect(
+      screen.getByText(/odpowiedź.*w pytaniu.*treść/i)
+    ).toBeInTheDocument();
+  });
+
+  it("should be possible to add question", async () => {
+    const { addQuestion, user, submit, fillFields } = setup();
+    let apiRequest: Quiz | null = null;
+
+    server.use(
+      http.post("*/quizzes", async ({ request }) => {
+        apiRequest = (await request.json()) as Quiz;
+        return HttpResponse.json({ ...apiRequest, id: "123" }, { status: 201 });
+      })
+    );
+
+    await fillFields();
+    await addQuestion();
+    await user.type(
+      screen.getAllByPlaceholderText(/treść pytania/i)[1],
+      "test question 2"
+    );
+    await user.type(
+      screen.getAllByPlaceholderText(/treść odpowiedzi/i)[2],
+      "test answer 3"
+    );
+    await user.type(
+      screen.getAllByPlaceholderText(/treść odpowiedzi/i)[3],
+      "test answer 4"
+    );
+    await submit();
+
+    expect(screen.getByText(/pytanie 2/i)).toBeInTheDocument();
+
+    expect(apiRequest!.questions).toHaveLength(2);
```

Comment:
zamiast używać non-null assertion zaleca się zassertowanie danej zmiennej, wtedy dostaniesz błąd w runtimie, a typescript będzie szczęśliwy


```suggestion
    import { assert } from "node:assert";
    
    assert.ok(apiRequest, "API request should be defined");
    expect(apiRequest.questions).toHaveLength(2);
```